c10/util/complex_math.h

```diff
@@ -316,6 +321,17 @@ C10_HOST_DEVICE inline c10::complex<T> log1p(const c10::complex<T>& z) {
     T z0 = std::hypot(x + 1, y);
     return {std::log(z0), theta};
   }
+#else
+  // Based on https://github.com/numpy/numpy/pull/22611#issuecomment-1667945354
+  c10::complex<T> u = z + T(1);
+  if (u == T(1)) {
+    return z;
+  } else if (u - T(1) == z) {
+    return log(u);
+  } else {
+    return log(u) * (z / (u - T(1)));
```

Comment:
Sadly these complex functions are far more costly than the real counterparts and it looks like the build is running out of memory when compiling them. A shot in the dark, there might be a compile time improvement by only calling `log(u)` once:
```cpp
auto log_u = log(u);
if (u - T(1) != z) {
    return log_u * (z / (u - T(1));
}
return log_u;
```